includes/board.h

```diff
@@ -13,18 +13,19 @@ class Board {
     };
 
     struct BoardCell {
-        Block& block;
+        Block* block;
         bool isOccupied;
-        BoardCell(Block& b, bool state = false) : block(b) {
-            this->isOccupied = state;
+        BoardCell() {
+            isOccupied = false;
+            block = new Block();
         }
-        void setOccupied() {
-            this->isOccupied = true;
+        BoardCell(Block& b, bool state = false) : block(&b) {
```

Comment:
this doesn't make sense to me. You are making a block reference equal to a block pointer

Reply:
nit: I personally don't like default parameters in constructors i.e. I'm a fan of making initialization state very explicit to the user of a class.

Reply:
nit: I also don't like how you named it state rather than occupied